### DISCUSSION TITLE
svgSelector `path` in V3

### DISCUSSION THREAD

**Discussion by cmahnke** [annotorious/annotorious#449] [General]
Hi Rainer,
I was looking for support of a svgSelector using `path`, but according to the source (and #400) it's not supported yet. I can't find any open issue for this. Is this still planned? Currently I think about using a release from the 2.x series.

**Comment by rsimon** [comment]
Hi @cmahnke ,

you should be able to display `path`s with Annotorious 2.7. (If you're looking for the source: 2.7 is [a separate branch](https://github.com/annotorious/annotorious/tree/2.x) in the repo.

Version 3 does not support paths. At least not yet. Adding this would still be significant work: you'd need to extend [the internal data model](https://annotorious.dev/guides/data-model/), extend the [W3C Adapter](https://annotorious.dev/guides/data-model/#the-w3c-adapter) (only if you need to use the W3C model), you'd need to add a `Path` shape [here](https://github.com/annotorious/annotorious/tree/main/packages/annotorious/src/annotation/shapes), and some more plumbing in various places.

If you want to use this with [OpenSeadragon](https://annotorious.dev/guides/openseadragon-iiif/), you'd also need to add the appropriate path rendering code. (Note that Annotorious OSD v3 is WebGL, not SVG.)

**Reply by cmahnke** [reply]
Thanks, I came to the same realization after looking into copy and past porting from 2.7
I found a solution to convert shapes into SVG `polygon`'s that also do the trick for me.

**Comment by cmahnke** [comment]
A bit related: It's currently not supported to have multiple SVG `polygon`'s in one selector.

**Reply by rsimon** [reply]
That's correct. Keep in mind that Annotorious __does not use the W3C model internally, nor does it use SVG__. (That's intentional :-)

In Annotorious' model, an [ImageAnnotation](https://github.com/annotorious/annotorious/blob/main/packages/annotorious/src/model/core/ImageAnnotation.ts) has exactly one [ImageAnnotationTarget](https://github.com/annotorious/annotorious/blob/03ee23fa905a3fc448b4aa51d769c15ccb31c93a/packages/annotorious/src/model/core/ImageAnnotation.ts#L10), which contains one [Shape](https://github.com/annotorious/annotorious/blob/main/packages/annotorious/src/model/core/Shape.ts). The Shape model is definitely going to evolve. There might be things like `MultiPolygon` in the future. But it will have to evolve along with the necessary drawing tools. (Because there's not much point in adding support for shape types for which there is no editing functionality, after all. So things will be going step by step.)

**Reply by cmahnke** [reply]
Yes, sure, I just came across this while decomposing SVG into multiple `polygon`'s, see above. I'm focuses on the W3C / SVG stuff since i'm working with SVG. That said, I'm not really intressted in the editing / drawing part. 

Thanks for your work!

**Reply by linuxbioinfoguy** [reply]
does the path object considered a "shape", for example, when use freehand to annotation, for some reason whenever a formatter is involved, just like the color formatter in the tutorial, if failed out once you draw the shape using the freehand tool, as in the image shows in the OSDAnnotationLayer.js line 695, as the selectedShape may be null, anybody could solve that ? 
![image](https://github.com/user-attachments/assets/648d557d-b958-414e-8909-a8c4fbbcc3f3)
